### DISCUSSION TITLE
Slow speeds compared to other containers/host

### DISCUSSION THREAD

**Discussion by sebcourant** [haugene/docker-transmission-openvpn#2741] [General]
I wrote an issue, which was understandably closed by @pkishino 
You can find it here for more information : #2740 

_**Here is the description of my problem :**_ 

Everything is technically working, and I haven't changed my config in ages.

Upgraded internet speed from 100mbps to 250mbps.

Noticed download speed in Transmission didn't improve.
Seems like I was limited the whole time.

What I checked :
- I was using :beta, reverted to :latest
- Made sure I was using latest version of Docker (and everything else on my host)
- Ran iperf3, pinging machines on my network : no bottleneck there I have gigabit lan
- Ran speedtest-cli several times, on the host directly, and inside containers on the same network bridge as transmission.
- Speedtest on the host and other containers on the bridge averaging 260mbps -> so not an issue with the bridge network.
- Speedtest on the transmission-openvpn container : average of 80mbps
- Checked which VPN server was used, tried on another device, speedtest there is also showing >250mbps so it's not provider related.

Talked to other people with same image, they also see speed heavily limited compared to what they can reach on other devices/containers

**Comment by gimlou** [comment]
Hi, same issue here, 

Do the same test, 10G cable, installed openspeedtest container on same docker network, Average 500Mbps UP/DOWN, stuck with transmission at 80Mbps (~10MBps), I'm looking into transmission container maybe

**Reply by sebcourant** [reply]
Hey, did you find a work around ? 
It seems this topic isn't very popular nobody is looking into it

**Reply by gimlou** [reply]
No, I don't. My only workaround is to use a traditional transmission container, and it works like a charm.

I noticed that maybe CPU usage is the key. Here are my test results: Same torrent file, same machine, same network.

Transmission + VPN: average 99% CPU usage
Transmission alone: average 25% CPU usage
VPN encryption consumes a lot, and with torrent encryption, I think a CPU bottleneck appears.

**Reply by sebcourant** [reply]
Wow that's a very high CPU usage, what device do you have ? I'm a bit surprised by that as it's not my experience.

I unfortunately don't want to use it without the VPN, as I need a reliable killswitch

**Comment by rassie** [comment]
I'm not sure whether this is the same problem, but I'm getting next to no speed after I've moved from Transmission 3.x + NordVPN installed straight on the host to Transmission 4.x + NordVPN installed in container. Same problem with `linuxserver`'s Transmission image in combination with `azinchen/nordvpn` container. Not sure whether I need to fix my Docker installation though.

**Comment by rassie** [comment]
Hmm. And just like that I've found what seems to be a solution for my problem. After setting Transmission's logging to `debug`, the log was full with failed IPv6 connections, even though IPv6 is disabled on host. After looking through the issue tracker, I've found issue https://github.com/transmission/transmission/issues/86 and PR https://github.com/transmission/transmission/pull/5329. Somewhat conviced this could be the solution, I've patched the image to build and use a nightly of Transmission and the problem went away. I do have other problems with it right now (container becomes unreachable after a couple of minutes), so I cannot recommend doing this, but there is hope for the upcoming Transmission 4.1.0, whenever that would be.

**Reply by gimlou** [reply]
I will investigate if I find time ;)

**Reply by rassie** [reply]
> container becomes unreachable after a couple of minutes

This was due to wrong (unwritable) download location set on torrents (taken over from non-docker times). Strategical "Set location" has helped a lot and now I don't have a problem anymore.

**Comment by gacpac** [comment]
So what was the solution? I'm having the same problem here

**Reply by sebcourant** [reply]
No solution found, I am not even sure the dev team or the contributors are looking into this.
I would like to at least understand the reason, it doesn't seem like an isolated issue on my end.

@pkishino do you know more ?

**Reply by gacpac** [reply]
i said while back, but i thought it was me only. I think is also got to do with the protocol being open vpn

**Reply by rassie** [reply]
Have you tried 4.0.5? There is a "Fixed 4.0.4 decreased download speeds for people who set a low upload bandwidth limit. `(https://github.com/transmission/transmission/pull/6134)`" item on the changelog. Maybe that's even what my problem was instead of IPv6.

**Reply by sebcourant** [reply]
Good find, even though I don't understand it, unfortunately. 
I'm simply using the images available on dockerhub (latest/edge), which seem to be on 4.0.4 if I'm not mistaken.

Maybe the creator of this PR, @tearfur knows more ?

**Reply by pkishino** [reply]
transmission 4.0.5 build is released so the build scripts for this container should pick it up some time today and make a build. I will make a "public" release later today

**Reply by sebcourant** [reply]
This is great news, thanks for the update !
I will update to 4.0.5 whenever possible and report back here

**Reply by pkishino** [reply]
Dev build is building now, once complete, please try it and report if this is solved

**Comment by gacpac** [comment]
I've been using dev build lol. Probably in the next couple days I'll test
again :D

**Comment by gacpac** [comment]
Yeah I just pulled the latest dev still showing 4.0.4 and same issues still

**Reply by sebcourant** [reply]
Yep, also pulled the latest from dockerhub and this problem isn't fixed.

**Comment by pkishino** [comment]
Please check that it is correct. Inside container try running
transmission-daemon -V
To see version.
If not 4.0.5 pull again..I suggest using the tagged releases :5.3.1 instead
of latest

**Reply by sebcourant** [reply]
I confirm I'm using 4.0.5

**Reply by pkishino** [reply]
Well, disappointing.. at this point I think it might still be better and you’ve also changed debug logs for any details as above?

**Comment by gacpac** [comment]
Wondering how do I change the debug logs?

**Reply by pkishino** [reply]
https://haugene.github.io/docker-transmission-openvpn/config-options/#debugging

**Comment by gacpac** [comment]
Thank you,

Na I don't have it

**Comment by kevle1** [comment]
![image](https://github.com/user-attachments/assets/eeb6590f-b40d-4db9-a3f9-16d30d89567c)

Old thread but in case anyone comes across this, try setting "Enable uTP" to false, or via the environment variable `TRANSMISSION_UTP_ENABLED` as false.

**Reply by voltechs** [reply]
> Old thread but in case anyone comes across this, try setting "Enable uTP" to false, or via the environment variable `TRANSMISSION_UTP_ENABLED` as false.

100% was the issue. Serves me right for "turn on all the settings for greater profit" without knowing what the setting does 😉

**Comment by gacpac** [comment]
That wasn't the case for me. I still can't figure it out for sure. Since
I'm not doing heavy lifting I don't mind much, but it's definetly annoying
not knowing why it happens
